File: examples/craft.lua
-- craft.lua — fire one of the Crafter Part's programmed Recipe Cards using
-- ingredients from the inventory on this channel.
--
-- Usage:
--   run craft.lua                   -- list every programmed slot + output
--   run craft.lua <slot>            -- craft once from that slot
--   run craft.lua <slot> <count>    -- craft <count> times
--
-- Layout this expects:
--   * one Adapter with a Crafter Part facing a vanilla crafting_table
--   * one inventory peripheral on the same channel holding the ingredients
--     (and where the crafted output ends up)

local args = { ... }
local slotArg = tonumber(args[1])
local countArg = tonumber(args[2]) or 1

local crafter = peripheral.find("crafter")
if not crafter then error("no crafter peripheral on this channel") end

if not slotArg then
    print("crafter: " .. crafter.name)
    for _, snap in ipairs(crafter:list()) do
        if snap then
            local out = snap.output or "(blank)"
            print(string.format("  [%2d] %s x%d", snap.slot, out, snap.outputCount))
        end
    end
    return
end

local source = inventory.find()
if not source then error("no inventory peripheral on this channel") end

local made = crafter:craft(slotArg, countArg, source)
print(string.format("crafted %d / %d", made, countArg))

File: examples/craft.js
// craft.js — JS twin of craft.lua. Fire one of the Crafter Part's programmed
// Recipe Cards using ingredients from the inventory on this channel.
//
// Usage:
//   run craft.js                  // list every programmed slot + output
//   run craft.js <slot>           // craft once from that slot
//   run craft.js <slot> <count>   // craft <count> times

(function () {
    var slotArg  = (args[0] != null) ? Number(args[0]) : null;
    var countArg = (args[1] != null) ? Number(args[1]) : 1;

    var crafter = peripheral.find('crafter');
    if (!crafter) throw new Error('no crafter peripheral on this channel');

    if (slotArg == null || isNaN(slotArg)) {
        print('crafter: ' + crafter.name);
        var snaps = crafter.list();
        for (var i = 0; i < snaps.length; i++) {
            var s = snaps[i];
            if (s != null) {
                var out = s.output != null ? s.output : '(blank)';
                print('  [' + s.slot + '] ' + out + ' x' + s.outputCount);
            }
        }
        return;
    }

    var source = inventory.find();
    if (!source) throw new Error('no inventory peripheral on this channel');

    var made = crafter.craft(slotArg, countArg, source);
    print('crafted ' + made + ' / ' + countArg);
})();
